controllers/admin/AdminCartRulesController.php

```diff
@@ -200,13 +200,35 @@ public function postProcess()
                 }
             }
 
-            // These are checkboxes (which aren't sent through POST when they are not check), so they are forced to 0
+            // These are checkboxes (which aren't sent through POST when they are not checked), so they are forced to 0
             foreach (['country', 'carrier', 'group', 'cart_rule', 'product', 'shop'] as $type) {
                 if (!Tools::getValue($type . '_restriction')) {
                     $_POST[$type . '_restriction'] = 0;
                 }
             }
 
+            // If the restriction is checked, but no item is selected, raise an error
+            foreach (['country', 'carrier', 'group', 'shop'] as $type) {
+                if (Tools::getValue($type . '_restriction') && empty(Tools::getValue($type . '_select'))) {
+                    switch ($type) {
+                        case 'country':
+                            $restriction_name = $this->trans('Country selection', [], 'Admin.Catalog.Feature');
+                            break;
+                        case 'carrier':
+                            $restriction_name = $this->trans('Carrier selection', [], 'Admin.Catalog.Feature');
+                            break;
+                        case 'group':
+                            $restriction_name = $this->trans('Customer group selection', [], 'Admin.Catalog.Feature');
+                            break;
+                        case 'shop':
+                        default:
+                            $restriction_name = $this->trans('Shop selection', [], 'Admin.Catalog.Feature');
+                            break;
+                    }
+                    $this->errors[] = $this->trans("'%s' is checked, but no item selected", [$restriction_name], 'Admin.Catalog.Notification');
```

Comment:
Hi @lmeyer1, 
I'm not sure to understand what is behind this variable. And what do you mean by "item"? Do you mean the restriction checkboxes?

Reply:
@Julievrz 
Logically, if someone selects the restriction checkbox, e.g. for Countries, but does not select any country, there is a problem.
Until now, in this case, the restriction checkbox was disregarded, and the cart rule applied to all countries.
But disregard the checkbox, even though it is checked was part of the bugs, I'm fixing here.
"'%s' is checked, but no item selected" becomes "'Country selection' is checked, but no item selected", where item is a country in the list.

Reply:
Thanks, it's clearer now. :-)